homeassistant/components/shelly/switch.py

```diff
@@ -60,90 +82,27 @@ async def async_setup_entry(
 ) -> None:
     """Set up switches for device."""
     if get_device_entry_gen(config_entry) in RPC_GENERATIONS:
-        return async_setup_rpc_entry(hass, config_entry, async_add_entities)
-
-    return async_setup_block_entry(hass, config_entry, async_add_entities)
-
-
-@callback
-def async_setup_block_entry(
-    hass: HomeAssistant,
-    config_entry: ConfigEntry,
-    async_add_entities: AddEntitiesCallback,
-) -> None:
-    """Set up entities for block device."""
-    coordinator = get_entry_data(hass)[config_entry.entry_id].block
-    assert coordinator
-
-    # Add Shelly Gas Valve as a switch
-    if coordinator.model == MODEL_GAS:
-        async_setup_block_attribute_entities(
-            hass,
-            async_add_entities,
-            coordinator,
-            {("valve", "valve"): GAS_VALVE_SWITCH},
-            BlockValveSwitch,
-        )
-        return
-
-    # In roller mode the relay blocks exist but do not contain required info
-    if (
-        coordinator.model in [MODEL_2, MODEL_25]
-        and coordinator.device.settings["mode"] != "relay"
-    ):
-        return
-
-    relay_blocks = []
-    assert coordinator.device.blocks
-    for block in coordinator.device.blocks:
-        if block.type != "relay" or is_block_channel_type_light(
-            coordinator.device.settings, int(block.channel)
-        ):
-            continue
-
-        relay_blocks.append(block)
-        unique_id = f"{coordinator.mac}-{block.type}_{block.channel}"
-        async_remove_shelly_entity(hass, "light", unique_id)
-
-    if not relay_blocks:
-        return
-
-    async_add_entities(BlockRelaySwitch(coordinator, block) for block in relay_blocks)
-
-
-@callback
-def async_setup_rpc_entry(
-    hass: HomeAssistant,
-    config_entry: ConfigEntry,
-    async_add_entities: AddEntitiesCallback,
-) -> None:
-    """Set up entities for RPC device."""
-    coordinator = get_entry_data(hass)[config_entry.entry_id].rpc
-    assert coordinator
-    switch_key_ids = get_rpc_key_ids(coordinator.device.status, "switch")
-
-    switch_ids = []
-    for id_ in switch_key_ids:
-        if is_rpc_channel_type_light(coordinator.device.config, id_):
-            continue
-
-        if coordinator.model == MODEL_WALL_DISPLAY:
-            if not coordinator.device.shelly.get("relay_in_thermostat", False):
+        rpc_coordinator = get_entry_data(hass)[config_entry.entry_id].rpc
+        if rpc_coordinator and rpc_coordinator.model == MODEL_WALL_DISPLAY:
+            if not rpc_coordinator.device.shelly.get("relay_in_thermostat", False):
                 # Wall Display relay is not used as the thermostat actuator,
                 # we need to remove a climate entity
-                unique_id = f"{coordinator.mac}-thermostat:{id_}"
+                unique_id = f"{rpc_coordinator.mac}-thermostat:0"
```

Comment:
Are we sure this will always be `0`?

Reply:
Device supports only 1 thermostat.

In any case, once all platforms will be on `EntityDescrition` this code will be moved to climate using a `removal_condition` that will take care of future devices with eventually more thermostats.